cloud/kubernetes/fluentd-configmap.yml

```diff
@@ -0,0 +1,102 @@
+# This ConfigMap is used to ingest logs against new resources like "k8s_container".
+# The configuration instruct Fluentd deamons to correctly parse CockroachDB logs,
+# extracting timestamp, severity and log message into the log output format for follow on output stanzas to process.
+# This is tested on Google Cloud Platform using a custom deployment of Stackdriver:
+# https://cloud.google.com/solutions/customizing-stackdriver-logs-fluentd
+
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: fluentd-config
+  namespace: kube-system
+  labels:
+    addonmanager.kubernetes.io/mode: Reconcile
+data:
+  containers.input.conf: |-
+    # Read and tail all log files on the node
+    # Match timestamp with one of the patterns, and prefix the tag with "reform".
+    <source>
+      @type tail
+      path /var/log/containers/*.log
+      pos_file /var/log/containers.log.pos
+      # Tags at this point are in the format of:
+      # reform.var.log.containers.<POD_NAME>_<NAMESPACE_NAME>_<CONTAINER_NAME>-<CONTAINER_ID>.log
+      tag reform.*
+      read_from_head true
+      format multi_format
+      <pattern>
+        format json
+        time_key time
+        time_format %Y-%m-%dT%H:%M:%S.%NZ
+      </pattern>
+      <pattern>
+        format /^(?<time>.+) (?<stream>stdout|stderr) [^ ]* (?<log>.*)$/
+        time_format %Y-%m-%dT%H:%M:%S.%N%:z
+      </pattern>
+    </source>
+
+    # This filter tries to parse the log records. The follow on filters will override the values set by this filter.
+    # This is for all containers that don't match the follow on filters - effectively treat this as the 'default' case.
+    # Note: Fluentd processes the log file on every 'Filter' that matches the pattern,
+    # however Fluentd allows only one 'Match' per pattern.
+    <filter reform.**>
+      @type parser
+      format /^(?<severity>\w)(?<time>\d{4} [^\s]*)\s+(?<pid>\d+)\s+(?<source>[^ \]]+)\] (?<log>.*)/
+      reserve_data true
+      suppress_parse_error_log true
+      emit_invalid_record_to_error false
+      key_name log
+    </filter>
+
+    # Reprocess the log line, this time match on particular formats.
+    # You can add your own custom format in this section.
+    # For CockroachDB, we are assigning the severity, time and log as capture groups.
+    <filter reform.**>
+      @type parser
+      key_name log
+      reserve_data true
+      suppress_parse_error_log true
+      emit_invalid_record_to_error false
+      format multi_format
+      <pattern>
+        # CockroachDB log format
+        format /^(?<severity>\w{1})(?<time>\d{6}\s\d{2}:\d{2}:\d{2})\.\d{6}\s*(?<log>.*)/
+        time_key time
+        time_format %y%m%d %H:%M:%S
```

Comment:
Why isn't this parsing the fractional seconds from the timestamp as is done above on lines 30 and 32 (`%Y-%m-%dT%H:%M:%S.%N`). I think we'd want this to be

```
time_format %y%m%d %H:%M:%S.%N
```

unless you have a particular reason in mind for not wanting the fractional components.

Reply:
Nope, updated :)